internal/command/binding_create.go

```diff
@@ -0,0 +1,317 @@
+	flags.StringVar(&service, "service", "", "Uses a Knative service as binding sink.")
+	flags.BoolVar(&force, "force", false, "Apply the changes even if the binding already exists.")
+	flags.StringArrayVar(&sourceProperties, "source-property", nil, `Add a source property in the form of "<key>=<value>"`)
+	return cmd
+}
+
+func createBinding(client camelkv1alpha1.CamelV1alpha1Interface, ctx context.Context, namespace string, options CreateBindingOptions) error {
+	kamelet, err := client.Kamelets(namespace).Get(ctx, options.Source, v1.GetOptions{})
+	if err != nil {
+		return knerrors.GetError(err)
+	}
+
+	if !isEventSourceType(kamelet) {
+		return fmt.Errorf("Kamelet %s is not an event source", options.Source)
+	}
+
+	sourceProps, err := parseProperties(options.SourceProperties)
+	if err != nil {
+		return knerrors.GetError(err)
+	}
+	sourceEndpoint := v1alpha1.Endpoint{
+		Properties: &sourceProps,
+		Ref: &corev1.ObjectReference{
+			Kind:       v1alpha1.KameletKind,
+			APIVersion: v1alpha1.SchemeGroupVersion.String(),
+			Name:       kamelet.Name,
+			Namespace:  kamelet.Namespace,
+		},
+	}
+
+	if err := verifyProperties(kamelet, sourceEndpoint); err != nil {
+		return knerrors.GetError(err)
+	}
+
+	var sinkRef corev1.ObjectReference
+	if options.Sink != "" {
+		sinkRef, err = decodeSink(options.Sink)
+	} else if options.Broker != "" {
+		sinkRef, err = decodeSink("broker:" + options.Broker)
+	} else if options.Channel != "" {
+		sinkRef, err = decodeSink("channel:" + options.Channel)
+	} else if options.Service != "" {
+		sinkRef, err = decodeSink("ksvc:" + options.Service)
+	} else {
+		err = fmt.Errorf("missing sink for binding - please use one of --sink, --broker, --channel, --service")
+	}
+
+	if err != nil {
+		return knerrors.GetError(err)
+	}
+
+	if sinkRef.Namespace == "" {
+		sinkRef.Namespace = namespace
+	}
+
+	sinkEndpoint := v1alpha1.Endpoint{
+		Ref: &sinkRef,
+	}
+
+	name := nameFor(options.Name, options.Source, sinkRef)
+
+	binding := v1alpha1.KameletBinding{
+		ObjectMeta: v1.ObjectMeta{
+			Namespace: namespace,
+			Name:      name,
+		},
+		Spec: v1alpha1.KameletBindingSpec{
+			Source: sourceEndpoint,
+			Sink:   sinkEndpoint,
+		},
+	}
+
+	existed := false
+	_, err = client.KameletBindings(namespace).Create(ctx, &binding, v1.CreateOptions{})
+	if err != nil && k8serrors.IsAlreadyExists(err) {
+		if options.Force {
+			existed = true
+
+			existing, err := client.KameletBindings(namespace).Get(ctx, binding.Name, v1.GetOptions{})
+			if err != nil {
+				return knerrors.GetError(err)
+			}
+			// Update the custom resource
+			binding.ResourceVersion = existing.ResourceVersion
+			_, err = client.KameletBindings(namespace).Update(ctx, &binding, v1.UpdateOptions{})
+			if err != nil {
+				return knerrors.GetError(err)
+			}
+		} else {
+			return fmt.Errorf("kamelet binding with name %q already exists", binding.Name)
+		}
+	}
+
+	if existed {
+		fmt.Printf("kamelet binding %q updated\n", name)
```

Comment:
In general I'd recommend to use `cmd.Out` and `cmd.Err` streams for printing out as this might be helpful in unit test and also if implementing options like `--quiet`